src/linear_algebra/Hermitian_inner_product.lean

```diff
@@ -0,0 +1,669 @@
+          { simpa using this} ,
+          { apply add_le_add (le_of_eq H) hle }}}
+    end}
+
+@[simp] lemma herm_norm_smul (a : ℂ) (x : α) : 
+|a • x| = abs(a)*|x| := 
+begin 
+  intros, 
+  dunfold herm_norm, 
+  rw [inprod_smul_left, inprod_smul_right, ←field.mul_assoc,
+    mul_conj, mul_re, of_real_im, zero_mul,
+    sub_zero, of_real_re, sqrt_mul (norm_sq_nonneg a)],
+  refl,
+end
+
+@[simp] lemma of_real_herm_norm_sqr (x : α) : 
+↑( |x|^2 ) = x ₀ x :=
+begin
+  dunfold herm_norm,
+  rw [sqr_sqrt (inprod_self_re_nonneg x), re_of_real (inprod_self_im x)],
+end
+
+@[simp] lemma of_real_herm_norm_mul_self (x : α) : 
+↑( |x|*|x| ) = x ₀ x :=
+begin
+  dunfold herm_norm,
+  rw [mul_self_sqrt (inprod_self_re_nonneg x), re_of_real (inprod_self_im x)],
+end
+
+noncomputable instance complex.metric_space : metric_space ℂ :=
+{ dist := λ x y, abs(x - y),
+  dist_self := by simp,
+  eq_of_dist_eq_zero := assume x y H, sub_eq_zero.mp (complex.abs_eq_zero.mp H),
+  dist_comm := begin intros, unfold dist, rw ←neg_sub, rw complex.abs_neg, end,
+  dist_triangle := abs_sub_le}
+
+noncomputable instance complex.normed_field : normed_field ℂ :=
+{ norm := abs,
+  dist_eq := by intros; refl, 
+  norm_mul := abs_mul,}
+
+noncomputable instance herm_inner_product_space.normed_space : 
+normed_space ℂ α := 
+{ norm := herm_norm,
+  dist_eq := by intros; refl,
+  norm_smul := herm_norm_smul}
+
+@[simp] lemma herm_norm_zero : 
+|(0 : α)| = 0 := @norm_zero α _
+
+lemma norm_smul_I {W : Type*} [normed_space ℂ W] (x : W) :
+∥I • x∥ = ∥x∥ :=
+begin
+  rw norm_smul,
+  unfold norm,
+  rw [abs_I, field.one_mul], 
+end
+
+@[simp] lemma herm_norm_smul_I (x : α) :
+|I • x| = |x| := norm_smul_I x 
+
+lemma norm_ne_zero_iff_ne_zero {G : Type*} [normed_group G] {g : G} : 
+∥g∥ ≠ 0 ↔ g ≠ 0 := 
+⟨ λ H, (iff_false_left H).mp (norm_eq_zero g), 
+  λ H, (iff_false_right H).mp (norm_eq_zero g)⟩ 
+
+theorem parallelogram_law (x y : α) :
+|x + y|^2 + |x - y|^2 = 2*|x|^2 + 2*|y|^2 :=
+begin
+  dunfold herm_norm,
+  rw [sqr_sqrt (inprod_self_re_nonneg (x + y)), sqr_sqrt (inprod_self_re_nonneg (x - y)),
+    sqr_sqrt (inprod_self_re_nonneg x), sqr_sqrt (inprod_self_re_nonneg y)],
+  suffices : (x ₀ x).re + ((x ₀ x).re + ((y ₀ y).re + (y ₀ y).re)) = 2 * (x ₀ x).re + 2 * (y ₀ y).re,
+  { rw [inprod_add_left, inprod_add_right, inprod_add_right,
+      inprod_sub_left, inprod_sub_right, inprod_sub_right,
+      add_re, add_re, sub_re, sub_re, sub_re],
+    simpa },
+  { rw [←conj_inprod y, conj_re],
+    ring }
+end
+
+lemma inprod_self_add (x y : α) :
+(x + y) ₀ (x + y) = (x ₀ x) + (y ₀ y) + (x ₀ y) + (y ₀ x) :=
+begin
+  rw [inprod_add_left, inprod_add_right, inprod_add_right],
+  simpa [field.add_assoc, field.add_comm],
+end
+
+def is_normalised (x : α) := |x| = 1
+
+noncomputable def normalise (x : α) := (↑|x|⁻¹ : ℂ) • x 
+
+lemma normalise_normalises (x : α) (ho : x ≠ 0) : 
```

Comment:
I would call this `normalise.is_normalised` (or with `s/s/z/g` if you want US spelling).